Framework/Algorithms/src/SampleCorrections/MayersSampleCorrection.cpp

```diff
@@ -134,9 +133,9 @@ void MayersSampleCorrection::exec() {
     params.sigmaSc = sampleMaterial.totalScatterXSection();
     params.cylRadius = radius;
     params.cylHeight = height;
-    MayersSampleCorrectionStrategy correction(params, inX, inputWS->readY(i),
-                                              inputWS->readE(i));
-    correction.apply(outputWS->dataY(i), outputWS->dataE(i));
+
+    MayersSampleCorrectionStrategy correction(params, inputWS->histogram(i));
+    correction.apply(outputWS->mutableY(i), outputWS->mutableE(i));
```

Comment:
Maybe apply could return a Histogram and you can use something like ouputWS->setHistogram(i, correction.apply());
You wouldn't then need the setSharedX above.